File: job-tests/numeros_primos.rb
class Alumno
  attr_reader :name

  def initialize(name)
    @name = name
  end

  def saluda
    puts "Hola, me llamo #{@name}. Mucho gusto!"
  end

  def dime_nums_primos(n)
    # numeros divisibles solo entre 1 y si mismos
    # no son divisibles entre ningun otro numero
    # loop y rev si i's hasta llenar 'n' cant primos
    # loop y rev si c/'i' es divisible entre alguno de los
    #   numeros primos anteriores, si termina y no, si es primo
    #   y se agrega ; Else, siguiente
    primos = [2]
    i = 3
    while primos.length < n
      # rev si el siguiente numero es menor divisible entre el/los
      # primos anteriores, si? i++, no y ya es el ultimo primo push
      primos.each_with_index do |p, j|
        # si i es divisible entre un numero primo[j] = p terminar loop
        # si Al_Final no? agregar al array de primos
        puts "i:#{i}, j:#{j}"
        break if i%p == 0
        # Si ya estamos en el final (y llego a esta linea) push a primos
        primos.push(i) if j == primos.length-1
      end
      i+=1
    end
    primos
  end
end

Fredy = Alumno.new('Fredy')
Fredy.saluda
p Fredy.dime_nums_primos(6); puts

class Egresado < Alumno
  def saluda
    super
    puts "Yo soy alumno egresado!!  (^_^)"
  end
end

Joss = Egresado.new('Joss')
Joss.saluda
p Joss.dime_nums_primos(4)

File: job-tests/numerosPrimos.js
class Alumno {
  constructor(name) {
    this.name = name;
  }

  saludo() {
    console.log(`Hola! Mi nombre es ${this.name}`);
  }

  dime_nums_primos(n) {
    // es primo si es divisible unicamente entre si mismo y 1
    // revisar si es divisible entre los numeros anteriores
    // Si lo es, no se agrega a la lista y se pasa a rev el sig
    // DEF primos Arr
    let primos = [2];
    // loop para revisar todos los numeros antes de la cantidad 'n' de No's Primos
    let i = 3;
    while(primos.length<n) {
      // loop para revisar si el numero 'i' es divisible entre los primos anteriores
      for(let j=0 ; j<primos.length ; j++) {
        // console.log(`i:${i}, j:${j}, primos:[${primos}]`);
        // i es divisible entre el primo[j]? ese i no es primo, terminar este loop
        // para revisar el siguiente i
        if(i%primos[j] == 0) break;
        // si ya revisamos todos y no fue divisible por ninguno es primo, se agrega
        if(j==primos.length-1) primos.push(i);
      }
      i++;
    }
    return primos;
  }
}

Jimmy = new Alumno('Jimmy');
Jimmy.saludo();
console.log(Jimmy.dime_nums_primos(6));
